File: BLL/PLANILLA.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BLL
{
    public class PLANILLA
    {
        public PLANILLA()
        {
            _planillaBE = new BE.PLANILLA();
        }
        
        #region property

        private BE.PLANILLA _planillaBE;
        //guarda un objeto BE.PLANILLA
        public BE.PLANILLA planillaBE
        {
            get { return _planillaBE; }
        }

        #endregion

        #region metodos

        //agrega un puntaje en la anotador de la planilla y cambia su estado
        public void puntuar(int posicion, int valor)
        {
            _planillaBE.AnotadorValores[posicion] = valor;
            _planillaBE.AnotadorEstado[posicion] = false;
        }

        //suma los puntajes actuales y devuelve el resultado
        public int resultado()
        {
            int total = 0;
            foreach (int i in _planillaBE.AnotadorValores)
            {
                total += i; 
            }
            return total;
        }

        #endregion






    }
}

File: BLL/PLANILLA.vb
Public Class PLANILLA

    Sub New()
        _planillaBE = New BE.PLANILLA
    End Sub

#Region "propertys"
    Private _planillaBE As BE.PLANILLA
    'guarda un objeto BE.PLANILLA
    Public ReadOnly Property planillaBE() As BE.PLANILLA
        Get
            Return _planillaBE
        End Get
    End Property

#End Region
    

#Region "Metodos"
    'agrega un puntaje en la anotador de la planilla y cambia su estado
    Public Sub puntuar(posicion As Integer, valor As Integer)
        _planillaBE.anotadorValor.Item(posicion) = valor
        _planillaBE.anotadorEstado.Item(posicion) = False
    End Sub

    'suma los puntajes actuales y devuelve el resultado
    Public Function resultado() As Integer
        Dim total As Integer = 0
        For Each i In _planillaBE.anotadorValor
            total += i
        Next
        Return total
    End Function
#End Region
    

    


End Class
